### PR TITLE
Resource descriptors are based on yaml files from the Helm charts

### PR DESCRIPTION
**Pull request by martonsereg**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes
| API breaks?     | yes
| Deprecations?   | no
| License         | Apache 2.0

### What's in this PR?
Instead of describing every resource in Go files, use the `yaml` descriptors from the Helm charts.

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)

### To Do
- [x] Track status of resources so they can be deleted during reconciliation if a config change makes them undesirable
- [ ] Watches are continuously triggered if `watch-created-resources-events` is enabled (maybe an issue with the Object Matcher lib)
- [ ] Multi cluster part is not working

### CODE REVIEW

File: scripts/download_charts.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+  # - remove the cleanup secrets job, we handle this in the installer
+  rm ${HELM_DIR}/istio/charts/security/templates/cleanup-secrets.yaml
+
+  # - we create custom resources in the normal way
+  rm ${HELM_DIR}/istio/charts/security/templates/create-custom-resources-job.yaml
+  rm ${HELM_DIR}/istio/charts/security/templates/configmap.yaml
+
+  # now make sure they're available
+  sed -i -e 's/define "security-default\.yaml\.tpl"/if and .Values.createMeshPolicy .Values.global.mtls.enabled/' ${HELM_DIR}/istio/charts/security/templates/enable-mesh-mtls.yaml
```
Comment (Laci21) [inline scripts/download_charts.sh]:
```suggestion
  sed -i '' -e 's/define "security-default\.yaml\.tpl"/if and .Values.createMeshPolicy .Values.global.mtls.enabled/' ${HELM_DIR}/istio/charts/security/templates/enable-mesh-mtls.yaml
```

File: scripts/download_charts.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  sed -i -e 's/define "security-permissive\.yaml\.tpl"/if and .Values.createMeshPolicy (not .Values.global.mtls.enabled)/' ${HELM_DIR}/istio/charts/security/templates/enable-mesh-permissive.yaml
```
Comment (Laci21) [inline scripts/download_charts.sh]:
```suggestion
  sed -i '' -e 's/define "security-permissive\.yaml\.tpl"/if and .Values.createMeshPolicy (not .Values.global.mtls.enabled)/' ${HELM_DIR}/istio/charts/security/templates/enable-mesh-permissive.yaml
```

File: scripts/download_charts.sh [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+  # remove namespace from the mutatingwebhook configuration because it's cluster scoped and setting the namespace provides reconciliation errors
+  sed -i -e '0,/namespace/{//d}' ${HELM_DIR}/istio/charts/sidecarInjectorWebhook/templates/mutatingwebhook.yaml
```
Comment (Laci21) [inline scripts/download_charts.sh]: This line gave me: `sed: 1: "0,/namespace/{//d}": extra characters at the end of d command`

Possible solution:

```suggestion
  sed -i '' -e '/namespace:/d' ${HELM_DIR}/istio/charts/sidecarInjectorWebhook/templates/mutatingwebhook.yaml
```

What do you think?

**Comment by martonsereg** [comment]
Closing this PR, Helm template based implementation will only be available in istio/operator

**Pull request closed without merging**